src/mechanics/constraintsPde/Constraints.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include <set>
+#include <eigen3/Eigen/Sparse>
+#include "mechanics/dofs/DofContainer.h"
+#include "mechanics/constraintsPde/Equation.h"
+
+namespace NuTo
+{
+namespace ConstraintPde
+{
+//! @brief stores constraint equations, separated by their dof type
+class Constraints
+{
+    using Equations = std::vector<Equation>;
+
+public:
+    //! @brief adds an equation
+    //! @param dof dof type
+    //! @param equation constraint equation
+    void Add(DofType dof, Equation equation);
+
+    //! @brief adds multiple equations
+    //! @param dof dof type
+    //! @param equations constraint equations
+    void Add(DofType dof, std::vector<Equation> equations);
+
+    //! @brief builds the time dependent constraint rhs vector for a specific dof type
+    //! @param dof dof type
+    //! @param time global time
+    //! @return constraint rhs vector
+    Eigen::VectorXd GetRhs(DofType dof, double time) const;
+
+    //! @brief builds a sparse matrix containing the constraint terms for a specific dof type
+    //! @param dof dof type
+    //! @param nIndependentDofs number of independent dofs for the dof type dof, required for a proper resize
+    //!        of the sparse matrix
+    //! @return sparse matrix containing the constraint terms where the last block with size (numDependent x
+    //! numDependent is removed)
+    Eigen::SparseMatrix<double> BuildConstraintMatrix(DofType dof, int numIndependentDofs) const;
```

Comment:
Name of the parameter `numIndependentDofs` is different in code and doxygen comment.